resource/opentelemetry-resource-detector-kubernetes/tests/test_kubernetes.py

```diff
@@ -0,0 +1,112 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from unittest.mock import mock_open, patch
+
+from opentelemetry.resource.detector.kubernetes import (
+    KubernetesResourceDetector,
+    get_kubenertes_pod_uid_v1,
+    get_kubenertes_pod_uid_v2,
+)
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.semconv.resource import ResourceAttributes
+
+MockKubernetesResourceAttributes = {
+    ResourceAttributes.K8S_POD_UID: "ecc2f8af-7742-4087-aeb1-4601bf25e1df",
+}
+
+
+class KubernetesResourceDetectorTest(unittest.TestCase):
```

Comment:
add a test case where you can check the expected_k8s_uid against the span.resource.k8s_uid.
- create a tracer_provider with  `resource=get_aggregated_resources(
                [
                    KubernetesResourceDetector(),
                ]
            ),`
- create a mock span and check if its resource attributes has k8s_uid and is it same a mock_k8s_uid?